docs/guides/src/main/server/configuring-database.adoc

```diff
@@ -0,0 +1,126 @@
+<#import "/templates/guide.adoc" as tmpl>
+<#import "/templates/kc.adoc" as kc>
+<#import "/templates/options.adoc" as opts>
+
+<@tmpl.guide
+    title="Configuring a database"
+    summary="An overview about how to configure relational databases"
+    includedOptions="db db-*">
+
+In this guide, you are going to understand how to configure the server to store data using different relational databases. You should also learn
+what are the databases supported by the server and how to proceed when using a unsupported database version.
+
+== Querying the list of supported databases
+
+The server has built-in support for different databases. You should be able to query the available databases by looking at the expected values
+for the `db` configuration option.
+
+The list of supported databases and their corresponding versions are:
+
+|===
+|Database | Version
+
+|mariadb| 10.3
+|mssql| 2016
+|mysql| 8
+|oracle| 12c
+|postgres| 10
+|===
+
+[NOTE]
+The server is opinionated about the databases and their respective versions. If you need support for a different database, there are some
+advanced configuration options to override the default configuration (e.g.: JDBC driver, Hibernate ORM Dialect, disable XA, etc). In this case, please help us by creating an issue to discuss
+the possibility to support additional databases.
+
+By default, the server uses the `h2-file` database. This is the default database that the server will use to persist data and
+really only exists so that you can run the authentication server out of the box. We highly recommend that you replace it with a more production ready external database. The `h2-file` database is not very viable in high concurrency situations and should not be used in a cluster either.
```

Comment:
```suggestion
really only exists so that you can run the authentication server out of the box.  The `h2-file` database is not suitable for production use-cases, and we highly recommend that you replace it with a production grade database.
```